streamalert/classifier/parsers.py

```diff
@@ -241,6 +241,10 @@ def _key_check(cls, record, schema, optionals=None, is_envelope=False):
         if not schema:
             return True
 
+        # Expect the record is a dict. Return False (schema doesn't match) if it is not.
+        if not isinstance(record, dict):
```

Comment:
I'd prefer if this, and the tests related to it, go into another PR if that's okay :)

Reply:
It is okay to move it to a separated PR. But we will have to create an separate SNS example topic for `trendmicro` in this PR. Or rebase this PR after the fix merged.
I am okay with both approach. @jack1902 your call.

Reply:
As this will fix both issues the quickest, I'd argue to merge it. Going through the trouble of removing a commit to rebase and carry out other work could introduce issues if things are forgotten. This will be the simplest way of resolving both at the same time

Reply:
Jack and me chatted over slack, we agree to move this to a separated PR! I will open an issue for this as well.